packages/cli/src/lib/server.ts

```diff
@@ -67,45 +69,57 @@ server.on('error', (err: Error) => {
   logger.error(`Server error: ${err.message}`, err)
 })
 
-loadDestination(targetDirectory).then(def => {
-  app.post(
-    '/:action',
-    asyncHandler(async (req: express.Request, res: express.Response) => {
-      const actionSlug = req.params.action
-
-      spinner.start(chalk`Handling ${process.env.DESTINATION}#${actionSlug} action request`)
-
-      try {
-        const destination = new Destination(def as CloudDestinationDefinition)
-        const action = destination.actions[actionSlug]
-
-        if (!action) {
-          const msg = `${destination.name} action '${actionSlug}' is invalid or not found`
-          spinner.fail(chalk`${msg}`)
-          return res.status(400).send(msg)
+loadDestination(targetDirectory)
+  .then((def) => {
+    app.post(
+      '/:action',
+      asyncHandler(async (req: express.Request, res: express.Response) => {
+        const actionSlug = req.params.action
+
+        spinner.start(chalk`Handling ${process.env.DESTINATION}#${actionSlug} action request`)
+
+        try {
+          const destination = new Destination(def as CloudDestinationDefinition)
+          const action = destination.actions[actionSlug]
+
+          if (!action) {
+            const msg = `${destination.name} action '${actionSlug}' is invalid or not found`
+            spinner.fail(chalk`${msg}`)
+            return res.status(400).send(msg)
+          }
+
+          await action.execute({
+            data: req.body.payload || {},
+            settings: req.body.settings || {},
+            mapping: req.body.payload || {},
+            auth: req.body.auth || {}
+          })
+
+          const debug = await getExchanges(destination.responses)
+          spinner.succeed(chalk`${destination.name} action '${actionSlug}' completed`)
+          return res.status(200).json(debug)
+        } catch (err) {
+          spinner.fail()
+          let statusCode = err?.status ?? 500
+          let msg = err?.message
+
+          if (err instanceof HTTPError) {
+            statusCode = err?.response?.status ?? 500
```

Comment:
Should we be falling back to the existing value instead in case `err.status` had a value?

```suggestion
            statusCode = err?.response?.status ?? statusCode
```